packages/cli/src/eventbus/MessageEventBusWriter/MessageEventBusLogWriter.ts

```diff
@@ -73,50 +79,52 @@ export class MessageEventBusLogWriter {
 	 *  First archives existing log files one history level upwards,
 	 *  then starts logging events into a fresh event log
 	 */
-	async startLogging() {
-		await MessageEventBusLogWriter.instance.getThread()?.startLogging();
+	startLogging() {
+		if (this.worker) {
+			this.worker.postMessage({ command: 'startLogging', data: {} });
+		}
 	}
 
 	/**
 	 *  Pauses all logging. Events are still received by the worker, they just are not logged any more
 	 */
 	async pauseLogging() {
-		await MessageEventBusLogWriter.instance.getThread()?.pauseLogging();
+		if (this.worker) {
+			this.worker.postMessage({ command: 'pauseLogging', data: {} });
+		}
 	}
 
 	private async startThread() {
 		if (this.worker) {
 			await this.close();
 		}
 		await MessageEventBusLogWriter.instance.spawnThread();
-		await MessageEventBusLogWriter.instance
-			.getThread()
-			?.initialize(
-				path.join(
-					MessageEventBusLogWriter.options.logBasePath,
-					MessageEventBusLogWriter.options.logBaseName,
-				),
-				MessageEventBusLogWriter.options.syncFileAccess,
-				MessageEventBusLogWriter.options.keepLogCount,
-				MessageEventBusLogWriter.options.maxFileSizeInKB,
-			);
+		if (this.worker) {
+			this.worker.postMessage({ command: 'initialize', data: MessageEventBusLogWriter.options });
+		}
 	}
 
 	private async spawnThread(): Promise<boolean> {
-		this.worker = await spawn<MessageEventBusLogWriterWorker>(
-			new Worker(`${parse(__filename).name}Worker`),
-		);
+		const parsedName = parse(__filename);
+		let workerFileName;
+		if (inTest) {
+			workerFileName = './dist/eventbus/MessageEventBusWriter/MessageEventBusLogWriterWorker.js';
```

Comment:
how is `threads` resolving the worker file name in tests?

Reply:
threads transpiles it via ts-node and then figures out the path using require.resolve. see https://github.com/andywer/threads.js/blob/master/src/master/implementation.node.ts